source/objects/waveClass.m

```diff
@@ -25,34 +25,34 @@
+            'option',                               3,...                   %
+            'depth',                                0, ...                  %
+            'direction',                            0, ...                  %
+            'speed',                                0)                      % (`structure`) Defines the current implementation. ``option`` (`integer`) Define the sub-surface current model to be used in WEC-Sim, options include: ``0`` for depth-independent model, ``1`` for 1/7 power law variation with depth, ``2`` for linear variation with depth, or ``3`` for no current. Default = ``3``, ``depth`` (`float`) Current depth [m]. Define the depth over which the sub-surface current is modeled. Must be defined for options ``1`` and ``2``. The current is not calculated for any depths greater than the specified current depth. Default = ``0``, ``direction`` (`float`) Current direction [deg]. Surface current direction defined using WEC-Sim global coordinate system. Default = ``0``, ``speed``  (`float`) Current seed [m/s]. Surface current speed that is uniform along the water column. Default = ``0``         
+        direction (1,:) double                      = 0;                    % (`float`) Incident wave direction(s) [deg]. Incident wave direction defined using WEC-Sim global coordinate system. Should be defined as a row vector for more than one wave direction. Default = ``0``
+        elevationFile (1,:) char                    = 'NOT DEFINED';        % (`char array`) Data file that contains the times-series data file. Default = ``'NOT DEFINED'``
+        gamma (1,:) double {mustBeScalarOrEmpty}    = [];                   % (`float`) Defines gamma, only used for ``JS`` wave spectrum type. Default = ``[]``        
+        height (1,:) double {mustBeNonnegative}     = [];                   % (`float`) Wave height [m]. Defined as wave height for ``regular``, or significant wave height for ``irregular``. Default =  ``'NOT DEFINED'``
+        marker (1,1) struct                         = struct(...            % (`structure`) Defines the wave marker. 
+            'location',                             [],...                  % 
+            'size',                                 10, ...                 %             
+            'style',                                1)                      % (`structure`) Defines the wave marker. `loc` (`nx2 vector`) Marker [X,Y] locations [m]. Default = ``[]``. ``size`` (`float`) Marker size in Pixels. Default = ``10``. ``style`` Marker style, options include: ``1``: Sphere, ``2``: Cube, ``3``: Frame. Default = ``1``: Sphere        
+        period (1,:) double {mustBeNonnegative}     = [];                   % (`float`) Wave period [s] . Defined as wave period for ``regular``, peak period for ``irregular``, or period of BEM data used for hydrodynamic coefficients for ``noWave``. Default = ``'NOT DEFINED'``

```

Comment:
@jtgrasb Another example where I believe the size should be (1,1). I tried within the bodyClass by setting it (1,1) and the issue was caught here if I used `waves.period = [7,8]`